core/templates/components/button-directives/exploration-embed-button-modal.component.spec.ts

```diff
@@ -0,0 +1,97 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for ExplorationEmbedButtonModalComponent.
+ */
+import { ComponentFixture, fakeAsync, TestBed, async} from
+  '@angular/core/testing';
+import { By } from '@angular/platform-browser';
+
+import { NgbActiveModal } from '@ng-bootstrap/ng-bootstrap';
+
+import { ExplorationEmbedButtonModalComponent } from
+  './exploration-embed-button-modal.component';
+import { SiteAnalyticsService } from 'services/site-analytics.service';
+
+class MockActiveModal {
+  dismiss(): void {
+    return;
+  }
+}
+
+class MockSiteAnalyticsService {
+  registerOpenEmbedInfoEvent(id: string): void {
+    return;
+  }
+}
+
+describe('ExplorationEmbedButtonModalComponent', () => {
+  let component: ExplorationEmbedButtonModalComponent;
+  let fixture: ComponentFixture<ExplorationEmbedButtonModalComponent>;
+  let siteAnalyticsService: SiteAnalyticsService;
+  let ngbActiveModal: NgbActiveModal;
+  beforeEach(async(() => {
```

Comment:
Why do we need `async` here? It is due to `compile components`?

Reply:
Yeah, I think so, at least angular's guide on testing had it this way so I just used it.